app/models/payment.rb

```diff
@@ -0,0 +1,21 @@
+module Payment
+  class << self
+    def table_name_prefix
+      'payment_'
+    end
+
+    def write_transaction(transaction:, provider:)
```

Comment:
The transaction class receives the store and calls `write_transaction` in `sale`:
```ruby
module Braintree
  class Transaction
    ...
    def sale
      if transaction.success? and @store
        @store.write_transaction(transaction: transaction, provider: :braintree )
      end

      transaction
    end
end
```

Looking at this, I see I'm not storing failed transactions!